src/main/java/hudson/plugins/buildblocker/BlockingJobsMonitor.java

```diff
@@ -176,15 +175,16 @@ private SubTask checkForRunningBuilds(Executor executor) {
                 task = ((MatrixConfiguration) task).getParent();
             }
 
-            AbstractProject project = (AbstractProject) task;
-
-            for (String blockingJob : this.blockingJobs) {
-                try {
-                    if (project.getFullName().matches(blockingJob)) {
-                        return task;
+            if (task instanceof Job) {
+                Job job = (Job) task;
+                for (String blockingJob : this.blockingJobs) {
+                    try {
+                        if (job.getFullName().matches(blockingJob)) {
+                            return job;
+                        }
+                    } catch (java.util.regex.PatternSyntaxException pse) {
+                        return null;
```

Comment:
Yes, it seems so. Just eat the exception and continue. BTW It's weird that any user did not hit this case before.